tests/specs/lemmas.k

```diff
@@ -244,6 +247,12 @@ module LEMMAS-HASKELL [symbolic, kore]
     imports EVM
     imports EDSL
 
+    //Simplification of double negation in term constraint
+    rule (B ==Bool false) ==Bool false => B                                                              [simplification]
+    //Simplifies { false #Equals EXP ==Bool false }
+    //see https://github.com/kframework/kore/issues/1944#issuecomment-657733745
+    rule  B ==Bool false               => notBool B                                                      [simplification]
+
```

Comment:
These rules will have a race condition. `(X ==Bool false) ==Bool false` may simplify down to either `B` (first rule), or to `notBool (X ==Bool false)`.

I think you should keep the second rule, and add a rule which says `notBool (X ==Bool false) => X`.

Reply:
The rules shouldn't have a race on their own, even if the Haskell backend is doing some magic (which it probably shouldn't be doing, but I'm not sure what magic you refer to).

Please make the rules presented here confluent, so that they work independent of what's going on in the backend.